### PR TITLE
Make ExecutableUnavailableError

### PR DESCRIPTION
**Pull request by alexrudy**
## PR Summary

When checking for executables which are optional dependencies, this makes matplotlib more forgiving of badly-behaved executables. Motivated by #14756, but should be generally helpful.

The new exception (`ExecutableUnavailableError`) inherits from `FileNotFoundError` to ensure that we don't break anyones existing code which uses `except FileNotFoundError:` to catch this kind of exception.

## PR Checklist

- [ ] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/__init__.py [hunk 1]
```diff
@@ -283,6 +283,12 @@ def wrapper():
 _ExecInfo = namedtuple("_ExecInfo", "executable version")
 
 
+class ExecutableUnavailableError(FileNotFoundError):
```
Comment (timhoffm) [inline lib/matplotlib/__init__.py]: Should this maybe be called `ExecutableNotFoundError` in analogy to `FileNotFoundError`?

Comment (alexrudy) [reply lib/matplotlib/__init__.py]: Yeah, that makes sense.

Comment (alexrudy) [reply lib/matplotlib/__init__.py]: I guess I went with `Unavailable` b/c the motivating case was when an executable `convert` is found, but `convert --version` returns nonsense.

**Comment by timhoffm** [comment]
@anntzer Not sure about this. I don't think we wrap all usages in a try-catch block. Therefore the ExecutableNotFoundError can rise to the user. In this case it should not be private.

**Comment by jklymak** [comment]
LGTM.  Not going to merge yet, in case anyone else wants to weigh in on the benefits of this.  I wasn't really aware that we could define our own errors like this, so am not sure if this is usual in packages...

**Comment by timhoffm** [comment]
@jklymak Refining excpetions by subclassing is a valid practice. See e.g. https://github.com/pandas-dev/pandas/blob/v0.24.2/pandas/errors/__init__.py#L96-L100 and surrounding code.

While I don't think one should do it for every exception, it's good if one can make it more specific (and maybe just want to catch that specific one). IMHO that's the case here.

File: lib/matplotlib/__init__.py [hunk 2]
```diff
@@ -283,6 +283,12 @@ def wrapper():
 _ExecInfo = namedtuple("_ExecInfo", "executable version")
 
 
+class ExecutableNotFoundError(FileNotFoundError):
+    """Error raised when an executable that Matplotlib optionally
+    depends on can't be found."""
+    pass
```
Comment (anntzer) [inline lib/matplotlib/__init__.py]: You don't need the "pass" here.

Also the docstring would typically be formatted
```
"""
Error raised...
"""
```
(basically either it fits in one line and we write
```
"""A one line docstring."""
```
or it doesn't and we write
```
"""
A multiline
docstring.
"""
```
or even
```
"""
One line, but just too long to have opening and closing quotes on the same line.
"""
```

**Pull request merged**